### PR TITLE
Add a blog post for mini site generation

### PR DESCRIPTION
**Pull request by abunashir**
We've recently added a new `site` interface to the CLI that aims to replace the old `Makefile`. This blog post explains how can we use this new interface to build a mini-site.

Fixes: #369

### CODE REVIEW

**Comment by abunashir** [comment]
Hello there, I've added a new blog post that explains how anyone can use the `metanorma site` interface to build a mini-site or customize it if necessary. I've tried to make it as less geeky as possible but could you guys please have a look at it and let me know your thoughts?

//cc: @ronaldtse

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    name: Abu Nashir
+    email: abunashir@gmail.com
+    social_links:
+      - https://github.com/abunashir
+
+excerpt: >-
+    Generate a collection mini site using Metanorma CLI
+---
+
+Each metanorma collection can be published as a mini site, there are couple of
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:16]: Don't refer to collections. A metanorma collection is already defined as a Metanorma document composed of other documents, something we are already doing with BIPM, and this functionality is documented as such in the metanorma-cli README. (We haven't quite gotten around to migrating that documentation to metanorma.com.)

What you're describing is a number of documents that are independent of each other, but can still be hosted on a single web directory. Call that a site, but refer to set of documents, not collection. (That set of documents can of course include a collection document!)

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+steps involed in this process to take a collection, compile it and generate the
+publishable site.
+
+To make this process easier, Metanorma has been offering a
+https://github.com/metanorma/mn-samples-itu/blob/master/Makefile[Makefile] that
+provides some targets to build the mini site. This has been working great until
+recently, but overtime it grew a lot and became harder to maintian, so we have
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:23]: maintian, attension, soruce: please check for typos.

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+|-- metanorma.yml                       # Configuration file in root directory
+----
+
+In root level, we have the `Gemfile`, `metanorma.yml` and `sources` directory.
+Inside `sources` there are two source files and an `images` folder for assets.
+
+==== Generate the site
+
+How can we generate the site now? Well it's pretty simple, all you need is the
+latest version of the https://github.com/metanorma/metanorma-cli[Metanorma CLI],
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:55]: Give the explicit version number: https://github.com/metanorma/metanorma-cli/releases/tag/v1.4.0

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+some specific files or add custom site metadata, well, then the next section is
+for you.
+
+=== Customizing site generation
+
+If you are like us and don't miss out the details then you might have seen the
+`-config` option in `site generate` help documentation, we didn't say anything
+earlier but that's actually the option to customize the build.
+
+This option takes a `YAML` file and usages that as site generation instructions,
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:91]: uses that

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+Currently it offers the follwing customiations:
+
+* Build a site with specific source files
+* Customize site metadata (Eg: title, and copyright)
+
+==== Build a site with specifc source files
+
+The demo projects contains couple of documents in the source directory, but for
+our example shake let's say we only want to use `sources/G.191.adoc` for the
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:100]: example's sake

**Review by opoudjis — CHANGES_REQUESTED** [review]

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc (hunk 2 above)
Comment (abunashir) [reply _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:23]: Thanks man, I will fix these

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc (hunk 3 above)
Comment (abunashir) [reply _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:55]: That's a good idea, but regarding the exact tag, I assume we would have some fixes coming up in the new releases and I wouldn't prefer to update the blog post every time, so should we point to - https://github.com/metanorma/metanorma-cli/releases instead?

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+----
+# filename: metanorma.yml
+
+metanorma:
+  collection:
+    name: Sample Name for the hero area
+    organization: Sample organization name
+----
+
+Then pass this configuration file as earlier to customize your build. You can
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:146]: https://github.com/metanorma/metanorma.com/issues/377 says to add metanorma.yml default config path to blog post....

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+    social_links:
+      - https://github.com/abunashir
+
+excerpt: >-
+    Generate a collection mini site using Metanorma CLI
+---
+
+A set of metanorma documents can be compiled and published as a min-site. There
+are a couple of steps involved in this process, like compiling each of these
+documents, building the collection, and generate the publishable site.
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:18]: Still would rather you avoid using "collection" completely. Replace with "set of documents" here

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+categories: about
+authors:
+  -
+    name: Abu Nashir
+    email: abunashir@gmail.com
+    social_links:
+      - https://github.com/abunashir
+
+excerpt: >-
+    Generate a collection mini site using Metanorma CLI
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:13]: Generate a mini site for a set of documents using Metanorma CLI

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+sample repo] as an exmaple, There are quite some files there but to keep the
+demo short let's pay attention to the following files and directory.
+
+[source,sh]
+----
+mn-samples-itu                          # Root directory
+|
+|-- sources                             # Source directory
+|   |-- images                          # Nested images for assets
+|   |-- G.191.adoc                      # Collection file inside sources
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:42]: `# Individual document file inside sources`

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+https://github.com/metanorma/metanorma-cli/releases/latest[Metanorma CLI] and
+then you can run the following:
+
+[source, sh]
+----
+metanorma site generate ./sources --output-dir site-folder
+----
+
+That's it, metanorma will take care of everything now, like compiling those
+source files, building the collection, and finally generating the HTML site.
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:65]: building the document set

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+name for the site copyright. To do that we can use the same configuration file
+and add the following data to it.
+
+[source, sh]
+----
+# filename: metanorma.yml
+
+metanorma:
+  collection:
+    name: Sample Name for the hero area
```
Comment (opoudjis) [inline _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:142]: ... Damn. Don't suppose we can change that?

**Review by opoudjis — DISMISSED** [review]

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc (hunk 6 above)
Comment (abunashir) [reply _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:155]: I've added these details in the earlier section of the same block.

File: _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc (hunk 11 above)
Comment (abunashir) [reply _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc:151]: That would be bit tricky for the moment, we already released it out and most of our existing repo were already using that structure before, but we are talking about restricting the YAML file anyway.

What about let's keep it like this for now, and as soon as we work on the restructuring then we will update the in another PR here?

Comment (ronaldtse) [reply _posts/2020-12-19-generate-mini-site-with-metanorma-cli.adoc]: I think this issue has been fixed, I've released a version without needing the `relaton` key.

**Pull request merged**